Source/CMakeLists.txt

```diff
@@ -13,11 +9,31 @@ if(CMAKE_SYSTEM_NAME MATCHES "Windows")
   add_definitions(-D_CRT_SECURE_NO_DEPRECATE)
 endif()
 
-# enable the latest C++ standard feature set,
-# and also disable MSVC specific extensions
-# to be even more standards compliant.
-check_and_add_flag(CPPLATEST /std:c++latest)
-check_and_add_flag(STANDARD_COMPLIANCE /permissive-)
+if(CMAKE_C_COMPILER_ID MATCHES "MSVC")
+  # enable the latest C++ standard feature set,
+  # and also disable MSVC specific extensions
+  # to be even more standards compliant.
+  check_and_add_flag(CPPLATEST /std:c++latest)
+  check_and_add_flag(STANDARD_COMPLIANCE /permissive-)
+else()
+  # Enable C++17, but fall back to C++14 if it isn't available.
+  # This would have been simpler if cmake supported CMAKE_CXX_STANDARD = 17;
+  # it currently doesn't, so this check has to be done manually.
```

Comment:
For CMake version 3.8 and up, `CMAKE_CXX_STANDARD` should be used, to correctly handle cases like https://gitlab.kitware.com/cmake/cmake/issues/16468

It's unfortunate they raise a hard error on an unsupported standard, rather than decaying to a previous one :/